File: app/src/main/java/com.example.safehome/System/SingletonSystem.java
package com.example.safehome.System;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


public class SingletonSystem {
	private static final Logger log = LoggerFactory.getLogger(SingletonSystem.class);
	private static System instance = null;

	public System getInstance() {
		return this.instance;
	}

	private SingletonSystem() {
		// TODO - implement SingletonSystem.SingletonSystem
		throw new UnsupportedOperationException();
	}

}

File: app/src/main/kotlin/com.example.safehome/System/SingletonSystem.kt
package com.example.safehome.System

import org.slf4j.LoggerFactory

class SingletonSystem private constructor() {
    val instance: System

    init {
        // TODO - implement SingletonSystem.SingletonSystem
        throw UnsupportedOperationException()
    }

    companion object {
        private val log = LoggerFactory.getLogger(SingletonSystem::class.java)
        private val instance: System? = null
    }
}